module_evaluations/TCR-15_2022-09-22.MD

```diff
@@ -0,0 +1,58 @@
+# ui-oa module acceptance criteria template
+
+## How to use this form
+When performing a technical evaluation of a module, create a copy of this document and use the conventions below to indicate the status of each criteria.  The evaluation results should be placed in the [module_evaluations](https://github.com/folio-org/tech-council/tree/master/module_evaluations) directory and should conform to the following naming convention:  `{JIRA Key}_YYYY-MM-DD.MD`, e.g. `TCR-1_2021-11-17.MD`.  The date here is used to differenciate between initial and potential re-evaluation(s).  It should be the date which the evaluation results file was created.
+
+* [x] ACCEPTABLE
+* [x] ~INAPPLICABLE~
+* [ ] UNACCEPTABLE
+  * comments on what was evaluated/not evaluated, why a criterion failed
+
+## [Criteria](https://github.com/folio-org/tech-council/blob/4dc5ac6d632fb880131cd214d212f973683b9ebe/MODULE_ACCEPTANCE_CRITERIA.MD)
+
+* [x] Upon acceptance, code author(s) agree to have source code canonically in folio-org github
+* [x] Copyright assigned to OLF
+* [x] Uses Apache 2.0 license
+* [ ] Third party dependencies use an Apache 2.0 compatible license
+  `@k-int/stripes-kint-components` does not contain a license file
+* [x] Module’s repository includes a compliant Module Descriptor
+* [ ] Modules must declare all consumed interfaces in the Module Descriptor “requires” and “optional” sections
+  `organizations` and `finance.exchange-rate` are absent in both `okapiInterfaces` and `optionalOkapiInterfaces`
+* [x] ~Environment vars are documented in the ModuleDescriptor - _note: read more at [https://wiki.folio.org/pages/viewpage.action?pageId=65110683](https://wiki.folio.org/pages/viewpage.action?pageId=65110683)_~
+* [x] ~Back-end modules must define endpoints consumable by other modules in the Module Descriptor “provides” section~
+* [x] ~All API endpoints are documented in RAML or OpenAPI~
+* [x] ~All API endpoints protected with appropriate permissions~
+* [ ] No excessive permissions granted to the module
+  In fact, no visible permissions are created, making it impossible to grant a user permission to access either the Open Access application or its settings. See UIOA-155.
```

Comment:
I think this might be one of the criteria that we may want to review and clarify.

During the mod-calendar review we discussed the potential wiggle room in our interpretations of criteria.

My interpretation of the reason for failing this criteria was that no user assignable permissions sets were defined, which I think is a separate aspect. To me, granting no permissions at all, meets the criteria that no excessive permissions are granted.

It seems to me that we have a hidden expectation about user assignable permission sets. Maybe we need a separate UI module specific criteria for defining appropriate user assignable permission sets?

Reply:
@marcjohnson-kint, yes, my understanding is that @jeremythuff and others have discussed a need to separate backend and frontend criteria. Yes, I agree that we need to reformulate this criterion, or add an additional one, in order to capture that necessary permissions are provided, in addition to avoiding excessive permissions being granted. 

As others noted when the TC reviewed this PR, we anticipated there would be a dialog between module submitters and the TC reviewers, hence failing this criterion in order to bring attention to the issue. Had this been the only missed criterion, it would have meant accepting a module that I knew to be completely unusable due to permissions. Personally, I don't think we would have been well-served by that.

Reply:
> @marcjohnson-kint, yes, my understanding is that @jeremythuff and others have discussed a need to separate backend and frontend criteria. Yes, I agree that we need to reformulate this criterion, or add an additional one, in order to capture that necessary permissions are provided, in addition to avoiding excessive permissions being granted.

Thank you for responding to my comment. I was unaware that this specific criteria had been discussed for reevaluation already.

> As others noted when the TC reviewed this PR, we anticipated there would be a dialog between module submitters and the TC reviewers, hence failing this criterion in order to bring attention to the issue. 

> Had this been the only missed criterion, it would have meant accepting a module that I knew to be completely unusable due to permissions. Personally, I don't think we would have been well-served by that.

I think that is a really interesting thought experiment.

I think it alludes to something really important in how we conduct these reviews, the balance between applying the policy / process as stated vs. as intended.

My understanding is that the TC agreed to apply the process and policy strictly in order to try to be as consistent as possible and because we wanted this process to be as apolitical as possible.

That was why I was uncomfortable with the evaluation I made of mod-calendar including a (towards acceptance) exception to the criteria and sought the TC view on that.

The TC then decided to accept modules without review (ui-plugin-find-authority?) by claiming responsibility for a gap in our policies (by assuming we intended to exclude plugins because we had set precedent for it previously).

As we continue to perform these evaluations, I think we need to be really mindful of both how we are doing so (it's not possible to remove all of our biases, which is why I excused myself from the OA module reviews) and what we are trying to achieve in doing them.

At this point (and this isn't specific to this review in any way, only brought to mind by your comment), I'm not confident that this process is achieving what we intend it to achieve.